src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,902 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System
+{
+    /// <summary>
+    /// represents dates with values ranging from January 1, 0001 Anno Domini (Common Era) through December 31, 9999 A.D. (C.E.) in the Gregorian calendar.
+    /// </summary>
+    public readonly struct DateOnly : IComparable, IComparable<DateOnly>, IEquatable<DateOnly>, IFormattable
+    {
+        private readonly int _dayNumber;
+
+        // Maps to Jan 1st year 1
+        private const int MinDayNumber = 0;
+
+        // Maps to December 31 year 9999. The value calculated from "new DateTime(9999, 12, 31).Ticks / TimeSpan.TicksPerDay"
+        private const int MaxDayNumber = 3652058;
+
+        private static int DayNumberFromDateTime(DateTime dt) => (int)(dt.Ticks / TimeSpan.TicksPerDay);
+
+        private DateTime GetEquivalentDateTime() => new DateTime(_dayNumber * TimeSpan.TicksPerDay);
+
+        private DateOnly(int dayNumber)
+        {
+            Debug.Assert((uint)dayNumber <= 3652058);
+            _dayNumber = dayNumber;
+        }
+
+        /// <summary>
+        /// Return the instance of the DateOnly structure representing the minimal possible date can be created.
+        /// </summary>
+        public static DateOnly MinValue { get; } = new DateOnly(MinDayNumber);
```

Comment:
❔ Does the JIT do a better job of inlining this, as opposed to:

```suggestion
        public static DateOnly MinValue => new DateOnly(MinDayNumber);
```

Reply:
No... This should use the `=>` version.

Reply:
No, this is different. your way will cause `new DateOnly(MinDayNumber);` execute every time. While `{ get; } = ` will just cache the value.

Reply:
> No, this is different. your way will cause `new DateOnly(MinDayNumber);` execute every time.

I was mostly asking as it relates to inlining, considering the constructor in use here takes no action aside from storing `MinDayNumber` to a field.

Reply:
According to [SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgMwAEu5AwuQN7HlPkD0LZ5AahADYCuccgDEAlglgAKAJTkAvAD4uvAQDoAggG5ijZh279BAZTgBjAPYA7ACbS5i/aoBCWogF8dTD+S8dYCPiYwSgZeDETMEeQADggiAG4Q8OQiFkEA+gkGLpHMXhEx8YmCDnASKUGZAjIK5BnKgrLklXDZOd7hbWwcWHgAbMEC5Gr05ADmcDAa5K5y5BZwAO4DpQAMUq05eboUPf0l5I52c4vLEmsb00A), the `=>` form produces this:

```
xor eax, eax
```

Reply:
CC @jkotas if has input on that.

Reply:
The constructor is trivial - just an assignment, it will always get inlined and morph into what is essentially an assignment of `0` to a local, which is ideal from the Jit's perspective as it will be able to reason about that local (constant folding, not double-zero, etc) and is essentially free.

In contrast, writing it as a `static readonly` struct field will be seen by the Jit as a load of "some" value from memory, which is worse (not to mention it necessitates a check for if the static constructor has run or not, which can only be elided for Tier1 codegen - and not always).

See also https://github.com/dotnet/runtime/issues/49036#issuecomment-790086369.